### PR TITLE
Fix: For developer tag, if the branch name has "/" change change by "…

### PR DESCRIPTION
**Pull request by lchico**
|Related issue|
|---|
https://github.com/wazuh/wazuh/issues/22969
||

## Description

Resolved issue with image tag naming during Docker uploads and package generation workflows by replacing '/' with '--' as a solution to ensure compliance with valid ASCII characters.

## Tests
change_tag
test_tag_2

### CODE REVIEW

File: .github/workflows/packages-build-wpk.yml [hunk 1]
```diff
@@ -105,7 +105,7 @@ jobs:
         if: (matrix.os == 'linux' && github.event.inputs.linux_reference != '') || (matrix.os == 'windows' && github.event.inputs.windows_reference != '') || (matrix.os == 'macos' && github.event.inputs.macos_reference != '')
         run: |
           if [ "${{ github.event.inputs.docker_image_tag }}" == "auto" ]; then echo "TAG=${{ env.VERSION }}" >> $GITHUB_ENV;
-          elif [ "${{ github.event.inputs.docker_image_tag }}" == "developer" ]; then echo "TAG=${{ github.ref_name }}" >> $GITHUB_ENV;
+          elif [ "${{ github.event.inputs.docker_image_tag }}" == "developer" ]; then echo "TAG=$(sed 's|[/\]|--|g' <<< ${{ github.ref_name }})" >> $GITHUB_ENV;
```
Comment (mjcr99) [inline .github/workflows/packages-build-wpk.yml:108]: Could we change the `github.event.inputs.docker_image_tag` context to `inputs.docker_image_tag` to keep consistency between files? I mean, remove the `github.event` references.

**Review by mjcr99 — CHANGES_REQUESTED** [review]

File: .github/workflows/packages-build-wpk.yml (hunk 1 above)
Comment (lchico) [reply .github/workflows/packages-build-wpk.yml]: Yes, sure. I also change all the cases.

**Comment by lchico** [comment]
Thanks, @mjcr99 . Comment addressed! PTAL

File: .github/workflows/packages-build-linux-agent.yml [hunk 2]
```diff
@@ -122,7 +122,7 @@ jobs:
         run: |
           VERSION=$(sed 's/^v\([0-9]*\.[0-9]*\.[0-9]*\)/\1/' $GITHUB_WORKSPACE/src/VERSION)
           if [ "${{ inputs.docker_image_tag }}" == "auto" ]; then echo "TAG=$VERSION" >> $GITHUB_ENV;
-          elif [ "${{ inputs.docker_image_tag }}" == "developer" ]; then echo "TAG=${{ github.ref_name }}" >> $GITHUB_ENV;
+          elif [ "${{ inputs.docker_image_tag }}" == "developer" ]; then echo "TAG=$(sed 's|[/\]|--|g' <<< ${{ github.ref_name }})" >> $GITHUB_ENV;
           else echo "TAG=${{ inputs.docker_image_tag }}" >> $GITHUB_ENV; fi
```
Comment (MarcelKemp) [inline .github/workflows/packages-build-linux-agent.yml:126]: Could we add the same filter for the case where the user sets his own tag?
Or if not, at least highlight in the description that it is not possible to use `/`.

**Review by MarcelKemp — APPROVED** [review]

**Pull request merged**